Orange/data/pandas_compat.py

```diff
@@ -170,16 +171,28 @@ def vars_from_df(df, role=None, force_nominal=False):
         _role = role
 
     # If df index is not a simple RangeIndex (or similar), put it into data
-    if not any(str(i).startswith('_o') for i in df.index) \
-            and not (df.index.is_integer() and (df.index.is_monotonic_increasing
-                                                or df.index.is_monotonic_decreasing)):
+    if (
+        # not range-like index - test first to skip slow startswith(_o) check
+        not (
+            df.index.is_integer()
+            and (df.index.is_monotonic_increasing or df.index.is_monotonic_decreasing)
+        )
+        # check that it does not contain Orange index
+        and (
+            # startswith is slow (for long drs) - firs check if col has strings
+            isinstance(df.index, pd.MultiIndex)
+            or not is_string_dtype(df.index)
+            or not any(str(i).startswith("_o") for i in df.index)
```

Comment:
Is it necessary to go through all the indexes? 
Some cases are not handled anyway, e.g:
`pd.DataFrame({"a": [1, 2], "b": [3, 4]}, index=["a", "o"])` vs. `pd.DataFrame({"a": [1, 2], "b": [3, 4]}, index=["_a", "_o"])`
Could this be speeded up even more by checking only, e.g., the first index (`df.index[0]`)?

Reply:
This one is actually fine. Since the argument to any is generator it will not be asked for the next value after first True appears.